prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,328 @@
+    }
+
+    /**
+     * Retreives the organization to send the emailschedule to
+     * 
+     * @param schedule the schedule to check organizations against
+     * @returns List of organizations for the schedule
+     */
+    private fun getOrgs( schedule: EmailSchedule ): List<String> {
+        return (
+            if (schedule.organizations.size > 0) 
+                schedule.organizations 
+            else 
+                fetchAllOrgs() 
+        )
+    }
+
+    /**
+     * Retrieves the list of all organization supported
+     * 
+     * @returns List of all organizations supported
+     */
+    private fun fetchAllOrgs(): List<String>{    
+        @Suppress( "NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER" )    
+        var ret = workflowEngine.db.transactReturning {  tx -> 
+            @Suppress( "UNRESOLVED_REFERENCE")
+            val settings = workflowEngine.db.fetchSettings( SettingType.ORGANIZATION, tx )
+            @Suppress( "NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER" )
+            settings.map{ it.getName() }
+        }
+       
+        @Suppress("OVERLOAD_RESOLUTION_AMBIGUITY")
+        System.out.println( ret );
+        return ret;
+    }
+
+
+    @FunctionName("createEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun createEmailSchedule(
+        @HttpTrigger(
+            name = "createEmailSchedule",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context );
+        var ret = request.createResponseBuilder(HttpStatus.CREATED);
+
+        if( user !== null ){
+            WorkflowEngine.databaseAccess.insertEmailSchedule(request.body, user );
+        }
+        else{
+            ret.status(HttpStatus.UNAUTHORIZED )
+        }
+        return ret.build();
+    }
+
+
+    fun validateUser( request: HttpRequestMessage<String?>, context: ExecutionContext ): String? {
+        var jwtToken = request.headers["authorization"] ?: ""
+
+        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""
+
+        if (jwtToken.isNotBlank()) {
+            try {
+                // get the access token verifier
+                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                    .setIssuer("https://hhs-prime.okta.com/oauth2/default")
+                    .build()
+                // get it to decode the token from the header
+                val jwt = jwtVerifier.decode(jwtToken)
+                    ?: throw Throwable("Error in validation of jwt token")
+                // get the user name and org
+                return jwt.claims["sub"].toString();
+            }
+            catch (ex: Throwable) {
+                context.logger.log(Level.WARNING, "Error in verification of token", ex)
+                return null
```

Comment:
Generally speaking, I prefer a "one way in, one way out" approach to functions. Here we have multiple exit points. I suggest something like the following specifically for the return-ing aspect of the code (which also sets your 'default' to be a good default:
```kotlin
        var jwtToken = request.headers["authorization"] ?: ""
        var ret = null

        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""

        if (jwtToken.isNotBlank()) {
            try {
                // get the access token verifier
                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
                    .setIssuer("https://hhs-prime.okta.com/oauth2/default")
                    .build()
                // get it to decode the token from the header
                val jwt = jwtVerifier.decode(jwtToken)
                    ?: throw Throwable("Error in validation of jwt token")
                // get the user name and org
                ret = jwt.claims["sub"].toString();
            }
            catch (ex: Throwable) {
                context.logger.log(Level.WARNING, "Error in verification of token", ex)
            }
        }
        return ret;
```

Reply:
Agree - done